dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/deploy/DefaultApplicationDeployer.java

```diff
@@ -691,7 +694,7 @@ private DynamicConfiguration prepareEnvironment(ConfigCenterConfig configCenter)
             } catch (Exception e) {
                 if (!configCenter.isCheck()) {
                     logger.warn("The configuration center failed to initialize", e);
-                    configCenter.checkOrUpdateInitialized(false);
+                    configCenter.setInitialized(false);
```

Comment:
As shown below, sequence number 1 will change `initialized` from false to true, but sequence number 2 cannot set `initialized` to false, so adjust to `configCenter.setInitialized(false);`
![image](https://user-images.githubusercontent.com/43363120/147056118-9ccefa0c-8d45-41dd-bd42-7a027a94ce12.png)